src/main.cpp

```diff
@@ -207,24 +203,37 @@ WUMS_DEINITIALIZE() {
 WUMS_APPLICATION_STARTS() {
     DEBUG_FUNCTION_LINE_VERBOSE("Inkay " INKAY_VERSION " starting up...\n");
 
-    // TODO - Add a way to reliably check this. We can't do it here since this path gets triggered before
-    // the plugin gets initialized.
-    //
-    // if (!Config::initialized && !Config::shown_uninitialized_warning) {
-    //     DEBUG_FUNCTION_LINE("Inkay module not initialized");
-    //     ShowNotification("Inkay module was not initialized. Ensure you have the Inkay plugin loaded");
-    //     Config::shown_uninitialized_warning = true;
-    // }
-
-    setup_olv_libs();
-    peertopeer_patch();
-    matchmaking_notify_titleswitch();
-    hotpatchAccountSettings();
+    // Reset plugin loaded flag
+    Config::plugin_is_loaded = false;
 }
 
-WUMS_APPLICATION_ENDS() {
+WUMS_ALL_APPLICATION_STARTS_DONE() {
+    if (!Config::internal_init_done) {
+        setup_olv_libs();
+        peertopeer_patch();
+        matchmaking_notify_titleswitch();
+        hotpatchAccountSettings();
+        Config::internal_init_done = true;
+    }
+    if (Config::initialized && !Config::plugin_is_loaded) {
+        DEBUG_FUNCTION_LINE("Inkay is running but the plugin got unloaded");
+        if (!Config::block_initialize) {
+            ShowNotification("Inkay module is still running. Please restart the console to disable Pretendo.");
```

Comment:
Same comment about Pretendo mode

```suggestion
            ShowNotification("Inkay module is still running. Please restart the console");
```